src/components/multiSelect/MultiSelect.tsx

```diff
@@ -0,0 +1,158 @@
+import React, { useState } from 'react';
+import Pill from '../Pill';
+import RemoveIcon from '../icons/RemoveIcon';
+
+/**Select component that supports single and multiple selections**/
+
+export type Option = {
+  label: string;
+  value: string;
+  key?: string;
+};
+interface SelectProps
+  extends React.DetailedHTMLProps<React.SelectHTMLAttributes<HTMLSelectElement>, HTMLSelectElement> {
+  /**
+   * Options to display in select
+   */
+  options: Option[];
+  /**
+   * Callback whenever selection changes
+   * @param val All selected values
+   * @param selectedIndex Index of the selection in display
+   */
+  notifySelection?: (val: string[], selectedIndex: number) => void;
+  /**
+   * Optional. If set, this value will be selected if that exist in option value
+   */
+  selectedValueProp?: string[];
+  /**
+   * Is Multiple selection or not, by default it is true
+   */
+  multiple?: boolean;
+}
+const defaultClassNames =
+  'block w-full py-2 px-3 border border-gray-300 bg-white rounded-md shadow-sm focus:outline-none focus:ring-sdb-100 focus:border-sdb-100 disabled:opacity-75 disabled:bg-gray-200 disabled:cursor-not-allowed';
+
+export const MultiSelect = React.forwardRef<HTMLSelectElement, SelectProps>(
+  ({ children, options, multiple = true, selectedValueProp, notifySelection, ...props }, ref) => {
+    const [selected, setSelected] = useState<Array<string>>([]);
+    const [selectionIndex, setSelectionIndex] = useState(-1);
+
+    React.useEffect(() => {
+      //Update selection, only if the given selection is different from what already exists
+      if (options.length < 0) return;
```

Comment:
Wouldn't `options.length == 0` be more useful here? I think `options.length < 0` is impossible.

Reply:
Yes, it's a mistake. Corrected